s2a/src/main/java/io/grpc/s2a/S2AChannelCredentials.java

```diff
@@ -58,13 +61,46 @@ public static Builder newBuilder(String s2aAddress) {
   public static final class Builder {
     private final String s2aAddress;
     private ObjectPool<Channel> s2aChannelPool;
-    private ChannelCredentials s2aChannelCredentials;
     private @Nullable S2AIdentity localIdentity = null;
+    private boolean useMtlsToS2A = false;
+    private @Nullable String privateKeyPath;
+    private @Nullable String certChainPath;
+    private @Nullable String trustBundlePath;
 
     Builder(String s2aAddress) {
       this.s2aAddress = s2aAddress;
       this.s2aChannelPool = null;
-      this.s2aChannelCredentials = InsecureChannelCredentials.create();
+    }
+
+    /**
+     * Sets whether to use mTLS to S2A. If true, the {@code privateKeyPath}, {@code certChainPath},
+     * and {@code trustBundlePath} must also be set.
+     */
+    @CanIgnoreReturnValue
+    public Builder setUseMtlsToS2A(boolean useMtlsToS2A) {
```

Comment:
Why not just let the caller set the ChannelCredentials to use? I'd understand this more if you were going to be ensuring certain security properties or modifying the configuration, but that doesn't seem the case.

Seems this entire thing is easier by just forcing the caller to pass ChannelCredentials to `newBuilder()`. That 1) avoids defaulting to insecure, 2) allows them to configure stronger security (e.g., there's some ways they could configure TLSv1.3-only), and avoids this error-prone API.

Compare:
```java
S2AChannelCredentials.newBuilder("metadata.google.internal", InsecureChannelCredentials.create())
    .build();
S2AChannelCredentials.newBuilder("something.googleapis.com", TlsChannelCredentials.newBuilder()
        .keyManager(certChainFile, privateKeyFile) // Safe if this was missing
        .trustManager(trustBundleFile) // Safe if this was missing
        .build())
    .build();
// -- vs --
// Not clear it is insecure
S2AChannelCredentials.newBuilder("metadata.google.internal").build();
// Oops! Actually insecure!
S2AChannelCredentials.newBuilder("something.googleapis.com")
    .setPrivateKeyPath(privateKeyPath)
    .setCertChainPaith(certChainPath)
    .setTrustBundlePath(trustBundlePath)
    .build();
```

From what I can tell this API is a convenience, but is less convenient.

Reply:
Done in https://github.com/grpc/grpc-java/pull/11544/commits/ab40adae0e60c3c5377e36c71fe5388c76a40172